pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -90,7 +89,7 @@ type connectivityCheckController struct {
 	enabledByDefault bool
 }
 
-type PodNetworkConnectivityCheckFunc func(ctx context.Context, syncContext factory.SyncContext) ([]*v1alpha1.PodNetworkConnectivityCheck, error)
+type PodNetworkConnectivityCheckFunc func(ctx context.Context, syncContext factory.SyncContext) ([]*applyconfigv1alpha1.PodNetworkConnectivityCheckApplyConfiguration, error)
```

Comment:
wouldn't it be easier to integrate if we were to take care of extracting the apply config from the PodNetworkConnectivityCheck object in the controller?

Reply:
> wouldn't it be easier to integrate if we were to take care of extracting the apply config from the PodNetworkConnectivityCheck object in the controller?

That will not handle a case of an explicit clearing of a particular field.  To express, "this field must be empty", you cannot use a struct.  That's one of the primary failings of understanding from controller-runtime as well.

Reply:
If the networking team is ok with never being able to unset a field, I suppose I'm ok creating that restriction if that's what they want, but I doubt that's what they intend in this case.

Reply:
I think connectivity check controller in [CNO](https://github.com/openshift/cluster-network-operator/blob/master/pkg/controller/connectivitycheck/connectivity_check_controller.go#L116) either generates same PodNetworkConnectivityCheck object or new PodNetworkConnectivityCheck object depending on target endpoint resides on same node or moved into different node. That way i hope apply config makes sense. I'll update CNO [PR](https://github.com/openshift/cluster-network-operator/pull/1649) with apply config to see of this works or not.